services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -416,6 +418,30 @@ testScimSideIsUpdated = do
         Scim.created meta `shouldBe` Scim.created meta'
         Scim.location meta `shouldBe` Scim.location meta'
 
+-- | Test that updating a user with the externalId of another user fails
+testUpdateToExistingExternalIdFails :: TestSpar ()
+testUpdateToExistingExternalIdFails = do
+    -- Create a user via SCIM
+    (tok, _) <- registerIdPAndScimToken
+    user <- randomScimUser
+    _ <- createUser tok user
+
+    newUser <- randomScimUser
+    storedNewUser <- createUser tok newUser
+
+    let userExternalId = Scim.User.externalId user
+    -- Ensure we're actually generating an external ID; we may stop doing this in the future
+    liftIO $ userExternalId `shouldSatisfy` isJust
+
+    -- Try to update the new user's external ID to be the same as 'user's.
+    let updatedNewUser = newUser{Scim.User.externalId = userExternalId}
+
+    env <- ask
+    -- Should fail with 409 to denote that the given externalId is in use by a
+    -- different user.
+    void $ updateUser_ (Just tok) (Just $ scimUserId storedNewUser) updatedNewUser (env ^. teSpar)
+            <!! const 409 === statusCode
```

Comment:
you can use `!!!` here and avoid the `void`.

Reply:
please also test this for creation: if use inject `user`'s `externalId` into `newUser` before calling `createUser`, it should also respond with `409`.  (i think.)

i would make the test case take a boolean to decide whether to test create or update.  less code duplication.

Reply:
This test case already exists for creation 😄

Reply:
oops, sorry.